Source/Core/Core/PowerPC/Jit64/JitAsm.cpp

```diff
@@ -103,35 +103,21 @@ void Jit64AsmRoutineManager::Generate()
   const bool assembly_dispatcher = true;
   if (assembly_dispatcher)
   {
-    // Fast block number lookup.
-    // ((PC >> 2) & mask) * sizeof(JitBlock*) = (PC & (mask << 2)) * 2
-    MOV(32, R(RSCRATCH), PPCSTATE(pc));
-    // Keep a copy for later.
-    MOV(32, R(RSCRATCH_EXTRA), R(RSCRATCH));
     u64 icache = reinterpret_cast<u64>(m_jit.GetBlockCache()->GetFastBlockMap());
-    AND(32, R(RSCRATCH), Imm32(JitBaseBlockCache::FAST_BLOCK_MAP_MASK << 2));
-    if (icache <= INT_MAX)
-    {
-      MOV(64, R(RSCRATCH), MScaled(RSCRATCH, SCALE_2, static_cast<s32>(icache)));
-    }
-    else
-    {
-      MOV(64, R(RSCRATCH2), Imm64(icache));
-      MOV(64, R(RSCRATCH), MComplex(RSCRATCH2, RSCRATCH, SCALE_2, 0));
-    }
+    MOV(32, R(RSCRATCH_EXTRA), PPCSTATE(pc));
```

Comment:
> Not necessary anymore, since you removed the `effectiveAddress` check.

I do need pc in RSCRATCH_EXTRA though when loading the block? Or can i use PPCSTATE(pc) there?

Reply:
Oh, sorry, I thought this line said `MOV(32, R(RSCRATCH_EXTRA), R(RSCRATCH));` like it did before this PR. We should keep this. (You could change it from `RSCRATCH_EXTRA` to `RSCRATCH` if you want to, but it doesn't really matter.)